factory/gen.go

```diff
@@ -67,40 +65,51 @@ func collectCNs(secret *v1.Secret) (domains []string, ips []net.IP, hash string,
 		}
 	}
 
-	hash = hex.EncodeToString(digest.Sum(nil))
 	return
 }
 
 func (t *TLS) Merge(target, additional *v1.Secret) (*v1.Secret, bool, error) {
-	return t.AddCN(target, cns(additional)...)
+	secret, updated, err := t.AddCN(target, cns(additional)...)
```

Comment:
This would make a great doc comment for the function itself and update this with this code intent.

Reply:
I added docs to the other exported functions in this method as well, hopefully that explains the flow a little better?